### ISSUE TITLE
Implement YOLOv8 training loop

### ISSUE THREAD

**Issue by uzFer** [uzFer/APS360-Project#6]
Implement YOLOv8 training loop, including checkpoint saving, testing accuracy, validation accuracy, output image logging, etc.
